src/util.rs

```diff
@@ -7,3 +7,64 @@ macro_rules! regex {
         &REGEX
     }};
 }
+
+use std::{
+    iter::FromIterator,
+    collections::HashMap,
+};
+
+
+/// Helper for `translate`, does the same thing as string.maketrans in python
+/// requires that f and t are the same length, or it takes the shorter one
+pub fn maketrans(f: &str, t: &str) -> HashMap<u8, char> {
+    HashMap::from_iter(f.bytes().zip(t.chars()))
+}
+
+/// Does effectively the same as string.translate in python, in 
+/// conjunction with `maketrans`, basically just replaces
+pub fn translate(s: String, table: HashMap<u8, char>) -> String {
```

Comment:
I believe we can just modified the String `s` here without the need to create a new String, that saves additional allocations by just performing the operations on the existing data. I wonder if `s.bytes().map(|byte| table.get(&byte).unwrap_or(byte))` works.